File: frontend/src/types/styles.ts
export type StyleType = 'professional' | 'casual' | 'friendly' | 'enthusiastic';

export interface StyleSettings {
  styleType: StyleType;
  intensity: number;
  culturalInfluence?: string;
  mannerisms?: string[];
}

export interface VideoGenerationRequest {
  image: File;
  audio?: File;
  text?: string;
  styleSettings?: StyleSettings;
}

export interface StylePreset {
  id: string;
  name: string;
  description: string;
  settings: StyleSettings;
  thumbnail: string;
}

File: backend/app/types/styles.py
from pydantic import BaseModel
from typing import List, Optional
from enum import Enum

class StyleType(str, Enum):
    PROFESSIONAL = "professional"
    CASUAL = "casual"
    FRIENDLY = "friendly"
    ENTHUSIASTIC = "enthusiastic"

class StyleSettings(BaseModel):
    styleType: StyleType
    intensity: int  # 1-10 scale
    culturalInfluence: Optional[str] = None
    mannerisms: Optional[List[str]] = None
